app/controllers/gitlab/webhook_base_controller.rb

```diff
@@ -0,0 +1,27 @@
+module Gitlab
+  class WebhookBaseController < ActionController::Base
+    before_action :validate_secret_key
+
+    def validate_secret_key
+      head :unprocessable_entity unless valid_secret_key?
+    end
+
+    private
+
+    def valid_secret_key?
+      request.headers['X-Gitlab-Token'] == integration.data['secret_key']
```

Comment:
The 3 private methods aren't being covered by the test
look: https://codeclimate.com/github/Codeminer42/cm42-central/app/controllers/gitlab/webhook_base_controller.rb/source?to_sha=f1f913eea86697c25343b61c5ac85c2e8a63a39e

Reply:
Will make it so the current test uses them aswell.